### PR TITLE
Implement GenesisChunked method

### PR DESCRIPTION
**Pull request by mauriceLC92**
Implements the GenesisChunked from issue #232

### CODE REVIEW

**Comment by adlerjohn** [comment]
Thank you for your contributor @mauriceLC92! Could you please fix the merge conflicts and mark this ready for review (if it's actually ready for review)?

**Comment by mauriceLC92** [comment]
> Thank you for your contributor @mauriceLC92! Could you please fix the merge conflicts and mark this ready for review (if it's actually ready for review)?

Sure thing! I need to add tests and then will mark it as ready for review :)

File: node/node.go [hunk 1]
```diff
@@ -147,6 +157,33 @@ func NewNode(ctx context.Context, conf config.NodeConfig, nodeKey crypto.PrivKey
 	return node, nil
 }
 
+func (n *Node) InitGenesisChunks(chunkSize int64) error {
```
Comment (mauriceLC92) [inline node/node.go:160]: Tendermint does not allow for this function to take in a chunk size: https://github.com/tendermint/tendermint/blob/master/internal/rpc/core/env.go#L152

I only did so in order to make writing the test a bit easier. The default chunk Size is quite big and cumbersome to test for if set to it's default: https://github.com/celestiaorg/optimint/pull/287/files#diff-375d57e386f20eaa5f09f02bb9d28bfc48ac3dca18d0325f59492208219e5618R45

Happy to change or hear thoughts

File: rpc/client/client_test.go [hunk 2]
```diff
@@ -75,6 +76,51 @@ func TestCheckTx(t *testing.T) {
...
+	wantId = 2
+	gc2, err := rpc.GenesisChunked(context.Background(), wantId)
+	assert.Error(err)
+	assert.Nil(gc2)
+
+	n2, _ := node.NewNode(context.Background(), config.NodeConfig{DALayer: "mock"}, privKey, proxy.NewLocalClientCreator(mockApp), &tmtypes.GenesisDoc{ChainID: "test"}, log.TestingLogger())
+	rpc2 := NewClient(n2)
+	gc2, err2 := rpc2.GenesisChunked(context.Background(), 0)
+	assert.Error(err2)
+	assert.Nil(gc2)
```
Comment (mauriceLC92) [inline rpc/client/client_test.go:121]: I would prefer to write these as separate tests and keep with a "one behaviour, one test" kind of approach to the tests. I see error cases are handled within one test in most cases though

Is this fine to leave as is?

File: node/node.go [hunk 3]
```diff
@@ -37,6 +39,12 @@ var (
 	indexerPrefix = []byte{2}
 )
 
+const (
+	// genesisChunkSize is the maximum size, in bytes, of each
+	// chunk in the genesis structure for the chunked API
+	genesisChunkSize = 16 * 1024 * 1024 // 16
```
Comment (adlerjohn) [inline node/node.go:45]: ?
```suggestion
	genesisChunkSize = 16 * 1024 * 1024 // 16 MiB
```

File: node/node.go [hunk 4]
```diff
@@ -147,6 +157,36 @@ func NewNode(ctx context.Context, conf config.NodeConfig, nodeKey crypto.PrivKey
 	return node, nil
 }
 
+// Creates a chunked format of the genesis document to make it easier to
+// iterate through larger genesis structures.
+// Default chunk size should be genesisChunkSize (16 * 1024 * 1024)
```
Comment (adlerjohn) [inline node/node.go:162]:
```suggestion
// InitGenesisChunks creates a chunked format of the genesis document to make it easier to
// iterate through larger genesis structures.
// Default chunk size should be genesisChunkSize
```

File: node/node.go (hunk 1 above)
Comment (adlerjohn) [reply node/node.go:163]: Wait, this `chunkSize` isn't actually used in the function body...

Comment (mauriceLC92) [reply node/node.go:163]: Ah you are right....thank you for catching that. I think I may have got a bit into a rabbit hole with the testing and thought my issue stemmed from the chunk size. Have updated to remove the `chunkSize` param now

File: node/node.go (hunk 3 above)
Comment (mauriceLC92) [reply node/node.go]: Thanks! Have added

File: node/node.go (hunk 4 above)
Comment (mauriceLC92) [reply node/node.go]: Updated 👍

**Comment by mauriceLC92** [comment]
@tzdybal the flaky [test](https://github.com/celestiaorg/optimint/issues/293) caused my build to fail again - are you able to re-run it?

**Review by tzdybal — APPROVED** [review]
Looks good :)

**Pull request merged**